mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -269,7 +271,11 @@ trait Params extends Identifiable with Serializable {
    * [[Param.validate()]].  This method does not handle input/output column parameters;
    * those are checked during schema validation.
    */
-  def validate(): Unit = validate(ParamMap.empty)
+  def validateParams(): Unit = {
```

Comment:
This implementation should not be necessary.  per-param validate is called whenever a ParamPair is constructed (and Params can only be specified via ParamPairs)